docs/Sources/tuist/tuist.docc/Tuist.md

```diff
@@ -86,6 +86,13 @@ The choice is yours. By adding .xcodeproj and .xcworkspace files to your .gitign
 - <doc:config>
 - <doc:workspace>
 
+### Extensions
+
+- <doc:generation-environment>
+- <doc:using-plugins>
```

Comment:
With the current content and structure I find it a bit hard to answer the following questions:
- What is a plugin?
- What can I do with a plugin?
- What types of plugin exist?

I'd organize the documentation to progressively answer the questions.
- **Extending Tuist**: Hey! did you know that you can extend Tuist.
- **Sharing extensions via plugins:** Hey! did you know that you can share your extensions through what we call plugins?

The former walks the user through the concept of extensibility in Tuist and the kinds of extensions. The latter takes that extensibility concept extend it to the idea of sharing them, which is what we call "plugin"

Reply:
Yeah, but Extending Tuist is a bunch of disjointed concepts:
- resource synthesizers
- tasks
- helpers
- templates

I don't think it makes sense to put them all into a single page. I also wanted to primarily migrate the docs, as that's the primary confusion.

We can improve them after.

Reply:
I'll take this on. Plugins is terribly confusing for someone coming across them for the first time.